packages/orbit-components/src/Itinerary/ItineraryPart/index.js

```diff
@@ -0,0 +1,73 @@
+// @flow
+import * as React from "react";
+import styled from "styled-components";
+
+import { ItineraryPartProvider } from "./context";
+import Stack from "../../Stack";
+import ItineraryPartStatus from "./components/ItineraryPartStatus";
+import getSpacingToken from "../../common/getSpacingToken";
+import defaultTheme from "../../defaultTheme";
+
+import type { Props } from ".";
+
+const StyledWrapper = styled.div`
+  position: relative;
+  margin-bottom: ${getSpacingToken};
+`;
+
+StyledWrapper.defaultProps = {
+  theme: defaultTheme,
+};
+
+const ItineraryPart = ({ status, label, children, spaceAfter, dataTest }: Props) => {
+  const content = React.Children.toArray(children);
+  const refs = React.useRef([]);
+  const [calculatedWidth, setCalculatedWidth] = React.useState(0);
+
+  React.useEffect(() => {
+    if (refs.current) {
+      const widths = refs.current.map(r => r.clientWidth).filter(Number);
+      setCalculatedWidth(Math.max(...widths));
+    }
+  }, [setCalculatedWidth]);
+
+  const parts =
+    content &&
+    content.length > 0 &&
+    React.Children.map(content, (el, i) => {
+      return (
+        <ItineraryPartProvider
+          index={i}
+          last={i === content.length - 1}
+          isNextHidden={content[i + 1] && content[i + 1].props.hidden}
+          count={content.length}
+          isHidden={el.props.hidden}
+          hasStatus={!!status}
+          calculatedWidth={calculatedWidth}
+        >
+          {el.type.displayName === "ItineraryPartPlace"
```

Comment:
I remember that we had problems in the past with usage of `displayName`.
https://github.com/kiwicom/orbit/pull/733/files#diff-79769f43c5aef4029a31e3e93d7dac9e5f2bae8e87782dc40952611f49994ef6L331

It won't work in some cases, or browsers, so the best would be to have different approach to it.

Reply:
I see that the ref is being used for re-calculation if there's some part has bigger width than the others. Since we want to cover this edge case I think that we can simply have a callback in the context that will be called after hydratation. The ref can be local for the `ItineraryPartPlace` or any other component and it will be trigger the state update in `useLayoutEffect`.

Reply:
good point about `useLayoutEffect` did not notice that flickering, so it's exactly the case for it, thanks :)